### PR TITLE
CMake: generate SupportedFeatures.json while building and install it into toolchain

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please build toolchain

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please build toolchain

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci Please smoke test Linux platform

**Comment by nkcsgexi** [comment]
@swift-ci Please clean test Linux platform

File: tools/driver/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,21 @@ target_link_libraries(swift-frontend
                         swiftSymbolGraphGen
                         LLVMBitstreamReader)
 
+# Generate and install supported feature JSON file to the toolchain.
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
+  set(DUMMY_SWIFT_INPUT "${SWIFT_RUNTIME_OUTPUT_INTDIR}/dummyInput.swift")
+  set(INT_FEATURE_FILE "${SWIFT_RUNTIME_OUTPUT_INTDIR}/SupportedFeatures.json")
+  add_custom_command(TARGET swift-frontend POST_BUILD
+                     COMMAND "touch" "${DUMMY_SWIFT_INPUT}"
```
Comment (compnerd) [inline tools/driver/CMakeLists.txt:21]: Can you use `${CMAKE_COMMAND} -E touch` instead please?

File: tools/driver/CMakeLists.txt [hunk 2]
```diff
@@ -13,6 +13,21 @@ target_link_libraries(swift-frontend
                         swiftSymbolGraphGen
                         LLVMBitstreamReader)
 
+# Generate and install supported feature JSON file to the toolchain.
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
```
Comment (compnerd) [inline tools/driver/CMakeLists.txt:17]: Why not-windows?  This condition seems wrong.  If this was due to the incorrect use of `touch`, this should be `if(NOT CMAKE_HOST_SYSTEM_NAME STREQUAL Windows)`, you could be building for Windows on macOS ;-)

**Review by compnerd — CHANGES_REQUESTED** [review]

File: tools/driver/CMakeLists.txt (hunk 2 above)
Comment (nkcsgexi) [reply tools/driver/CMakeLists.txt:17]: Yeah, we should use `CMAKE_HOST_SYSTEM_NAME` here.

File: tools/driver/CMakeLists.txt (hunk 1 above)
Comment (nkcsgexi) [reply tools/driver/CMakeLists.txt]: Fixed.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: tools/driver/CMakeLists.txt (hunk 2 above)
Comment (compnerd) [reply tools/driver/CMakeLists.txt]: I think that question of why not-windows still remains.  What makes SupportedFeatures.json not applicable to Windows?

**Pull request merged**